src/main/java/com/conveyal/analysis/controllers/NetworkTileController.java

```diff
@@ -0,0 +1,158 @@
+
+import static com.conveyal.analysis.controllers.GtfsVectorTileMaker.clipScaleAndSimplify;
+import static com.conveyal.analysis.util.HttpStatus.OK_200;
+import static com.conveyal.analysis.util.HttpUtils.CACHE_CONTROL_IMMUTABLE;
+import static com.conveyal.r5.common.GeometryUtils.floatingWgsEnvelopeToFixed;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Defines HTTP API endpoints to return Mapbox vector tiles of the street network. This is not simply the underlying
+ * OSM data, but the street edges created from that OSM data. This allows the user to verify how the OSM tags have
+ * been interpreted as mode permissions, speeds, etc. and eventually should allow visualizing the effets of scenarios
+ * on the street nework, includng scenarios that change the characteristics of street edges or create new streets.
+ * See GtfsVectorTileMaker for more information on the vector tile spec and tile numbers.
+ */
+public class NetworkTileController implements HttpController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(NetworkTileController.class);
+
+    /**
+     * Complex street geometries will be simplified, but the geometry will not deviate from the original by more
+     * than this many tile units. These are minuscule at 1/4096 of the tile width or height.
+     */
+    private static final int LINE_SIMPLIFY_TOLERANCE = 5;
+
+    /** The zoom level at which each StreetClass appears, indexed by StreetClass.code from 0...4. */
+    private static final int[] zoomForStreetClass = new int[] {8, 10, 11, 12, 13};
+
+    private final TransportNetworkCache transportNetworkCache;
+
+    public NetworkTileController (TransportNetworkCache transportNetworkCache) {
+        this.transportNetworkCache = transportNetworkCache;
+    }
+
+    @Override
+    public void registerEndpoints (Service sparkService) {
+        // Ideally we'd have region, project, bundle, and worker version parameters.
+        // Those could be path parameters, x-conveyal-headers, etc.
+        // sparkService.get("/api/bundles/:gtfsId/vectorTiles/:x/:y/:z", this::getTile);
+        // Should end in .mvt but not clear how to do that in Sparkframework.
+        //         sparkService.get("/:bundleId/:modificationNonceDigest/tiles", this::buildIndex, JsonUtil.toJson);
+        sparkService.get("/:bundleId/tiles", this::buildIndex, JsonUtil.toJson);
+        sparkService.get("/:bundleId/tiles/:z/:x/:y", this::getTile);
+    }
+
+    private TransportNetwork getNetworkFromRequest (Request request) {
+        // "bundleId" is also the "graphId" in an `AnalysisWorkerTask`
+        final String bundleId = request.params("bundleId");
+        // final String modificationNonceDigest = request.params("modificationNonceDigest");
+        checkNotNull(bundleId);
+        TransportNetwork network = transportNetworkCache.getNetwork(bundleId);
+        // "61fe817974f6230b0363aae1-8c07ddd4f8bd29ac10a4a109dd27d7b58dabd56c" // elevation only
```

Comment:
These appear to be hard coded testing IDs. Should they be removed?

Reply:
Yes, just left over from experiments and prototyping.